afb-types/src/chmgr-types.rs

```diff
@@ -10,165 +10,184 @@
-#[serde(rename_all = "lowercase")]
-pub enum PlugState {
-    PlugIn,
-    Lock,
-    Error,
-    PlugOut,
-    Unknown,
-}
-
-#[derive(Serialize, Deserialize, Debug, Clone, Copy)]
-#[serde(rename_all = "lowercase")]
-pub enum IsoState {
-    Iso20,
-    Iso2,
-    Iso3,
-    Iec,
-    Unset,
-}
-
-AfbDataConverter!(charging_event, ChargingMsg);
-#[derive(Serialize, Deserialize, Debug, Clone)]
-#[serde(rename_all = "lowercase")]
-pub enum ChargingMsg {
-    Plugged(PlugState),
-    Power(PowerRequest),
-    Iso(IsoState),
-    Auth(AuthMsg),
-    State(ChargingState),
-    Reservation(ReservationStatus)
-}
-
-AfbDataConverter!(reservation_state, ReservationState);
-#[derive(Serialize, Deserialize, Debug, Clone)]
-#[serde(rename_all = "lowercase")]
-pub struct ReservationState {
-    pub id: i32,
-    pub start: Duration,
-    pub stop: Duration,
-}
-
-AfbDataConverter!(charging_state, ChargingState);
-#[derive(Serialize, Deserialize, Debug, Clone)]
-#[serde(rename_all = "lowercase")]
-pub struct ChargingState {
-    #[serde(skip)]
-    pub updated: bool,
-    #[serde(skip)]
-    pub reservation: Option<ReservationState>,
-    pub imax: u32,
-    pub pmax: u32,
-    pub plugged: PlugState,
-    pub power: PowerRequest,
-    pub iso: IsoState,
-    pub auth: AuthMsg,
-
-}
-
-impl ChargingState {
-    pub fn default() -> Self {
-        ChargingState {
-            updated: false,
-            imax: 32, // Fulup TBD should comme from energy mgr
-            pmax: 22,
-            plugged: PlugState::Unknown,
-            power: PowerRequest::Idle,
-            iso: IsoState::Unset,
-            auth: AuthMsg::Idle,
-            reservation: None,
-        }
-    }
-}
-
-AfbDataConverter!(charging_actions, ChargingAction);
-#[derive(Serialize, Deserialize, Debug, Default)]
-#[serde(rename_all = "lowercase", tag = "action")]
-pub enum ChargingAction {
-    #[default]
-    READ,
-    SUBSCRIBE,
-    UNSUBSCRIBE,
-}
-pub enum ReservationAction {
-    NOW,
-    DELAY,
-    CANCEL,
-}
-
-AfbDataConverter!(charging_msg, ReservationStatus);
-#[derive(Serialize, Deserialize, Debug, Clone, Copy)]
-#[serde(rename_all = "lowercase")]
-pub enum ReservationStatus {
-    Accepted,
-    Refused,
-    Pending,
-    Cancel,
-    Request,
-}
-
-AfbDataConverter!(reservation_session, ReservationSession);
-#[derive(Serialize, Deserialize, Debug, Clone)]
-#[serde(rename_all = "lowercase")]
-pub struct ReservationSession {
-    pub id: i32,
-    pub tagid:String,
-    pub start: Duration,
-    pub stop:  Duration,
-    pub status: ReservationStatus,
-}
-
-pub fn chmgr_registers() -> Result<(), AfbError> {
-    charging_actions::register()?;
-    plug_state::register()?;
-    charging_state::register()?;
-    error_state::register()?;
-    power_request::register()?;
-    charging_event::register()?;
-    reservation_session::register()?;
-    reservation_state::register()?;
-    power_limit::register()?;
-
-    Ok(())
-}
+ use crate::prelude::*;
```

Comment:
I do not know exactly why, but you have one whitespace on each line. This breaks a bit the "diff" view.

Reply:
Sorry, I think when I copy the the modified content of charging-binding/chmgr-types into josev-binding/chmgr-types, the lines must have shifted. Do you know how I can fix it if needed ?

Reply:
If you use VS Code, "Format Document" (ctrl+shift+I) should help. Or `rustfmt` from the command line. This should be done on all files you modify before submitting a PR.